components/net/filemanager_thread.rs

```diff
@@ -550,30 +608,41 @@ impl <UI: 'static + UIProvider> FileManagerStore<UI> {
 
     fn promote_memory(&self, blob_buf: BlobBuf,
                        sender: IpcSender<Result<SelectedFileId, BlobURLStoreError>>, origin: FileOrigin) {
-        match Url::parse(&origin) { // parse to check sanity
-            Ok(_) => {
-                let id = Uuid::new_v4();
-                self.insert(id, FileStoreEntry {
-                    origin: origin.clone(),
-                    file_impl: FileImpl::Memory(blob_buf),
-                    refs: AtomicUsize::new(1),
-                    // Valid here since PromoteMemory implies URL creation
-                    is_valid_url: AtomicBool::new(true),
-                });
-
-                let _ = sender.send(Ok(SelectedFileId(id.simple().to_string())));
-            }
-            Err(_) => {
-                let _ = sender.send(Err(BlobURLStoreError::InvalidOrigin));
-            }
+        // Check origin format correctness
+        if Url::parse(&origin).is_ok() || origin == "file://" {
```

Comment:
Because `Url::parse` is not happy with an origin equal to `file://`. It is kinda a hack (related to `get_blob_origin`, `parse_blob_url` etc. as well)